File: 프로그래머스/lv2/42578. 위장/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> m = new HashMap<String, Integer>();
        
        for(String[] it : clothes) {
            m.put(it[1], m.getOrDefault(it[1], 0) + 1);
        }
        
        for(int it : m.values()) {
            answer *= (it + 1);
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/lv2/42578. 위장/위장.kt
class Solution {
    fun solution(clothes: Array<Array<String>>): Int {
        var answer = 1
        
        var m = clothes.groupBy{it[1]}.toList()
        m.forEach {
            answer *= (it.second.size + 1)
        }
    
        
        return answer - 1
    }
}